repair_order_group/models/repair_order.py

```diff
@@ -0,0 +1,222 @@
+            if repairs_to_process:
+                repairs_to_process.with_context(skip_group_sync=True).write(
+                    {"partner_id": vals["partner_id"]}
+                )
+        return res
+
+    def _action_repair_confirm(self):
+        """Confirm repair and cascade confirmation to other group members."""
+        res = super()._action_repair_confirm()
+        if not self.env.context.get("skip_group_confirm"):
+            repairs_to_process = self.mapped("grouped_repair_ids").filtered(
+                lambda r: r.state == "draft"
+            )
+            if repairs_to_process:
+                repairs_to_process.with_context(
+                    skip_group_confirm=True
+                )._action_repair_confirm()
+
+        return res
+
+    def action_repair_cancel(self):
+        """Cancel repair and cascade cancellation to other group members."""
+        res = super().action_repair_cancel()
+        if not self.env.context.get("skip_group_cancel"):
+            repairs_to_process = self.mapped("grouped_repair_ids").filtered(
+                lambda r: r.state not in ("cancel", "done")
+            )
+            if repairs_to_process:
+                repairs_to_process.with_context(
+                    skip_group_cancel=True
+                ).action_repair_cancel()
+
+        return res
+
+    def action_create_sale_order(self):
+        """Create Sale Order - single one for group or standard behavior if no group.
+
+        For repair groups: creates one sale order containing all valid repairs
+        from the same group. Maintains core Odoo validations while extending
+        functionality for grouped repairs.
+
+        Returns:
+            dict: Action to view the created sale order
+        """
+        self.ensure_one()
+
+        # Core Odoo validations
+        if any(repair.sale_order_id for repair in self):
+            concerned = self.filtered("sale_order_id")
+            ref_str = "\n".join(ro.name for ro in concerned)
+            raise UserError(
+                _(
+                    "You cannot create a quotation for a repair order that is "
+                    "already linked to an existing sale order.\n"
+                    "Concerned repair order(s):\n%(ref_str)s",
+                    ref_str=ref_str,
+                )
+            )
+
+        if any(not repair.partner_id for repair in self):
+            concerned = self.filtered(lambda ro: not ro.partner_id)
+            ref_str = "\n".join(ro.name for ro in concerned)
+            raise UserError(
+                _(
+                    "You need to define a customer for a repair order in order to "
+                    "create an associated quotation.\n"
+                    "Concerned repair order(s):\n%(ref_str)s",
+                    ref_str=ref_str,
+                )
+            )
+
+        # Standard behavior for non-grouped repairs
+        if not self.group_id:
```

Comment:
How is this aligned with the code above? Why is this done only here and not before?